scala/private/rule_impls.bzl

```diff
@@ -240,18 +243,25 @@ StatsfileOutput: {statsfile_output}
   ctx.actions.write(
       output = argfile, content = scalac_args + optional_scalac_args)
 
+  scalac_provider = ctx.attr._scalac[_ScalacProvider]
```

Comment:
@johnynek given your work trying to make `compile_scala` more explicit and less `ctx` dependent; Would you rather this be a parameter?